src/core/ddsi/src/q_entity.c

```diff
@@ -5914,17 +6376,21 @@ int delete_proxy_writer (struct ddsi_domaingv *gv, const struct ddsi_guid *guid,
   local_reader_ary_setinvalid (&pwr->rdary);
   GVLOGDISC ("- deleting\n");
   builtintopic_write (gv->builtin_topic_interface, &pwr->e, timestamp, false);
+#ifdef DDS_HAS_TYPE_DISCOVERY
+  /* Unref tl_meta before removing from entity index, because
```

Comment:
I think I get the reasoning; the bit I am not yet sure of is what the consequences of unnecessarily triggering an update for the endpoint matching would be. I think it would simply be a waste of CPU cycles. Am I right?

Reply:
Yes that's correct, and this will not cause an extra trigger of the endpoint matching. This only ensures that the proxy endpoint that's being deleted will not be registered with the type anymore, so that `ddsi_tl_handle_reply` will not try to do any matching for this proxy endpoint.